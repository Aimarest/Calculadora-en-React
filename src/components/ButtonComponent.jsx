

export const ButtonComponent = ({value, action, isDisable}) => {

  const handleClick = () =>{
    action(value);
  }

  return (
    <button onClick={handleClick}  disabled={isDisable} name={value} value={value} className="btn">{value}</button>
  )
}
