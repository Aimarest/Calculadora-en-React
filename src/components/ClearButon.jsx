

export const ClearButon = ({ value, resetResult }) => {

  const handleClick = () => {
    resetResult();
  }
  return (
    <button onClick={handleClick} value={value} className="btnClear btn">{value}</button>
  )
}
