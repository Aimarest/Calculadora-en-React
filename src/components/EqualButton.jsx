

export const EqualButton = ({value, action, isDisable,setIsDisable}) => {


    const handleClick = () =>{
        action(value);
        setIsDisable(true);
        
      }
    
      return (
        <button onClick={handleClick} value={value} disabled={isDisable} className="btn">{value}</button>
      )
}
