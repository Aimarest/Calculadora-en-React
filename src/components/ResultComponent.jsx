

export const ResultComponent = () => {
  return (
    <>
        <h2 className="result">Result</h2>
    </>
  )
}
