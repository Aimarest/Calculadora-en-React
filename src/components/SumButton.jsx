

export const SumButton = ({value,  sum }) => {

    const handleSum = () =>{
        sum(value)
    }
  return (
   <button onClick={handleSum} value={value} className="sumBtn">{value}

   </button>
  )
}
