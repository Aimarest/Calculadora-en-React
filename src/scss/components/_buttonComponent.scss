.buttons {
    display: grid;
    grid-template-columns: 1fr 1fr 1fr 1fr;
    grid-template-rows: 1fr 1fr 1fr 1fr;
    gap: 3px 3px;

    @media screen and (min-width:400px) {

        gap: 10px 15px;

    }

    @media screen and (min-width:764px) {

        gap: 15px 20px;

    }

}

.btn:active {
    background-color: #00916e;
}

.btn {
    display: flex;
    align-items: center;
    justify-content: center;
    border-radius: 32px;
    padding: 0.6em;
    border: none;
    cursor: pointer;
    min-width: 2.2em;
    font-size: 1em;
    color: white;
    background-color: #00916e;
    box-shadow: 5px 5px 10px -3px #00000040, -5px -5px 15px 3px #00ffc11f;

    @media screen and (min-width:400px) {

        padding: 1.3em
    }

    @media screen and (min-width:764px) {

        padding: 1.8em
    }

    @media screen and (min-width:1024px) {

        padding: 2em
    }

}

.btnClear {
    margin-top: 1em;
    background-color: #224870;
    width: 8em;

    @media screen and (min-width:764px) {

        margin-top: 3em;

    }

    @media screen and (min-width:1024px) {
        width: 20em;
        margin-top: 5em;

    }

}