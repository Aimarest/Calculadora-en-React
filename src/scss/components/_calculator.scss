.calculator {
    margin: 2em;
    display: flex;
    flex-direction: column;
    justify-content: center;
    align-items: center;
    width: 40%;
    max-height: 30%;
    max-width: 500px;
    background-color: $calculator-color;
    border-radius: 12px;
    padding: 1.5em 2.5em;
    border: 4px solid $background-color;
    box-shadow: 15px 10px 0 5px $shadow;

    @media screen and (min-width:400px) {
        padding: 4em 3.8em;
    }

    @media screen and (min-width:764px) {
        padding: 6em 2.5em;
    }
}