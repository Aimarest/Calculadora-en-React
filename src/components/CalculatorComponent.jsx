import { ButtonComponent } from './ButtonComponent';
import { ResultComponent } from './ResultComponent';
import {useState} from 'react';

export const CalculatorComponent = () => {
const [text, setText] = useState('');

  const addValueToResult = (value) =>{
    setText((text) => [...text, value + ''])
  }
  return (
    <>
      <ResultComponent text={text} />
      <section className='buttons'>
        <ButtonComponent value={7} addValueToResult={addValueToResult} />
        <ButtonComponent value={8} addValueToResult={addValueToResult}/>
        <ButtonComponent value={9} addValueToResult={addValueToResult}/>
        <ButtonComponent value={'%'} addValueToResult={addValueToResult}/>
        <ButtonComponent value={'Clear'} addValueToResult={addValueToResult}/>
        <ButtonComponent value={4}addValueToResult={addValueToResult} />
        <ButtonComponent value={5} addValueToResult={addValueToResult}/>
        <ButtonComponent value={6} addValueToResult={addValueToResult}/>
        <ButtonComponent value={'X'}addValueToResult={addValueToResult} />
        <ButtonComponent value={'/'} addValueToResult={addValueToResult}/>
        <ButtonComponent value={1} addValueToResult={addValueToResult}/>
        <ButtonComponent value={2} addValueToResult={addValueToResult}/>
        <ButtonComponent value={3} addValueToResult={addValueToResult}/>
        <ButtonComponent value={'+'} addValueToResult={addValueToResult}/>
        <ButtonComponent value={'='} addValueToResult={addValueToResult}/>
        <ButtonComponent value={0} addValueToResult={addValueToResult}/>
      </section>

    </>
  )
}
