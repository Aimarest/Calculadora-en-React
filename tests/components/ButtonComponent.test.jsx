import { ButtonComponent } from "../../src/components/ButtonComponent";
import { render, screen, fireEvent } from '@testing-library/react';
describe('Test on the ButtonComponent', () => {
    const initialValue = 2;
    const isDisabled = false;
    const action = jest.fn()
    test('The initial component should be displayed', () => {
        const { container } = render(<ButtonComponent value={initialValue} disabled={isDisabled} action={action} />
        );
        expect(container).toMatchSnapshot();
    })
    test('The function should be called when pressing the button', () => {
        render(<ButtonComponent value={initialValue} disabled={isDisabled} action={action} />
        );
        const btn = screen.getByRole('button');
        fireEvent.click(btn);
        expect(action).toHaveBeenCalled();
    })
})
