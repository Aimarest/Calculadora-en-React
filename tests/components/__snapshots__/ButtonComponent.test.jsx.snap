// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test on the ButtonComponent The initial component should be displayed 1`] = `
<div>
  <button
    class="btn"
    value="2"
  >
    2
  </button>
</div>
`;
